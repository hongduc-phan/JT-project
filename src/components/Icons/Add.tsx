import React, {SVGAttributes} from 'react';

const Notification = (props: SVGAttributes<HTMLOrSVGElement>) => (
  <svg width="1em" height="1em" viewBox="0 0 24 24" {...props}>
    <g fill="none" fillRule="evenodd">
      <path
        fill="currentColor"
        fillRule="nonzero"
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
      />
      <path d="M0 0h24v24H0z" />
    </g>
  </svg>
);

export default Notification;
