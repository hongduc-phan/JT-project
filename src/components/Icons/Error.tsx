import React, {SVGAttributes} from 'react';

const Error = (props: SVGAttributes<HTMLOrSVGElement>) => (
  <svg width="1em" height="1em" viewBox="0 0 24 24" {...props}>
    <g fill="none" fillRule="evenodd">
      <path d="M0 0h24v24H0z" />
      <path
        fill="currentColor"
        fillRule="nonzero"
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
      />
    </g>
  </svg>
);

export default Error;
