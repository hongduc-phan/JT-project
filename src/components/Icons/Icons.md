```js
const {
  Edit,
  Delete,
  Setting,
  EPayRoll,
  ETime,
  ArrowDown,
  ArrowLeft,
  ArrowRight,
  ArrowUp,
  Notification,
  Account,
  Add,
  Search,
  Error,
} = require('./');

<div className="icon-list">
  <div>
    <div>
      <Edit />
    </div>
    <div>
      <Delete />
    </div>
    <div>
      <Setting />
    </div>
    <div>
      <Setting active={true} />
    </div>
    <div>
      <EPayRoll />
    </div>
    <div>
      <EPayRoll active={true} />
    </div>
    <div>
      <ETime />
    </div>
    <div>
      <ETime active={true} />
    </div>
    <div>
      <ArrowDown />
    </div>
    <div>
      <ArrowLeft />
    </div>
    <div>
      <ArrowUp />
    </div>
    <div>
      <ArrowRight />
    </div>
  </div>
  <div>
    <div>
      <Notification />
    </div>
    <div>
      <Account />
    </div>
    <div>
      <Add />
    </div>
    <div>
      <Search />
    </div>
    <div>
      <Error />
    </div>
  </div>
</div>;
```
