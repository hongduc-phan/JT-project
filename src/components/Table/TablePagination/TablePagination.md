Example

```js
const TablePagination = require('./').default;

<TablePagination
  labelRowsPerPage="Items per page:"
  rowsPerPageOptions={[5, 10, 25]}
  page={1}
  count={200}
  rowsPerPage={10}
/>;
```
