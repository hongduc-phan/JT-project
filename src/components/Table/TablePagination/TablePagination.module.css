.wrapper {
  display: flex;
  align-items: center;
  justify-content: flex-end;
  margin-top: 1rem;
}

.textFieldRoot {
  width: 3.5rem;
  margin-left: 0.5rem;
}

.textField {
  padding: 0.125rem 0;
  border: none;
  border-radius: 0;
}

.textFieldFocus {
  border-bottom: 0.125rem solid var(--blue-default);
}

.inputSelect {
  font-size: 0.75rem;
  min-height: 1.25rem;
  line-height: 1.67;
}

.textFieldArrow {
  right: 0;
  color: var(--blue-default);
}

.labelDisplayedRows {
  margin-left: 2.25rem;
  margin-right: 2.75rem;
}
