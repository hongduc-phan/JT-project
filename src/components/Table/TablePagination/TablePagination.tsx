import React, {useState} from 'react';

import MenuItem from '../../MenuItem';
import Typo, {TypoVariants, TypoColors} from '../../Typo';
import TextField from '../../TextField';

import Button, {ButtonVariants} from '../../Button';
import {ArrowLeft, ArrowRight} from '../../Icons';

import styles from './TablePagination.module.css';

export interface ITablePaginationProps {
  labelRowsPerPage?: string;
  page: number;
  count: number;
  rowsPerPage: number;
  rowsPerPageOptions?: number[];
  onChangePage?: (page: number) => void;
  onChangeRowsPerPage?: (value: number) => void;
}

const labelDisplayedRows = (from: number, to: number, count: number) => {
  return `${from}-${to} of ${count}`;
};

const TablePagination = ({
  labelRowsPerPage,
  page,
  count,
  rowsPerPage,
  rowsPerPageOptions = [10, 25, 50, 100],
  onChangePage,
  onChangeRowsPerPage,
}: ITablePaginationProps) => {
  const [maxPage, setMaxPage] = useState(Math.ceil(count / rowsPerPage));

  const handleOnClickPreviousButton = () => {
    if (onChangePage) {
      let newPage = page - 1;
      newPage = newPage < 1 ? 1 : newPage;
      onChangePage(newPage);
    }
  };

  const handleOnClickNextButton = () => {
    if (onChangePage) {
      let newPage = page + 1;
      newPage = newPage > maxPage ? maxPage : newPage;
      onChangePage(newPage);
    }
  };

  const handleOnChangeRowsPerPage = (value: number) => {
    if (onChangeRowsPerPage) {
      setMaxPage(Math.ceil(count / value));
      onChangeRowsPerPage(value);
    }
  };

  return (
    <div className={styles.wrapper}>
      {rowsPerPageOptions.length > 1 && (
        <Typo variant={TypoVariants.subTitle} color={TypoColors.greyMedium}>
          {labelRowsPerPage}
        </Typo>
      )}
      {rowsPerPageOptions.length > 1 && (
        <TextField
          select={true}
          value={rowsPerPage}
          className={styles.textFieldRoot}
          classes={{
            textField: styles.textField,
            textFieldFocus: styles.textFieldFocus,
            inputSelect: styles.inputSelect,
            arrow: styles.textFieldArrow,
          }}
          onChangeSelectValue={handleOnChangeRowsPerPage}
        >
          {rowsPerPageOptions.map((item, index) => (
            <MenuItem key={index} value={item}>
              {item}
            </MenuItem>
          ))}
        </TextField>
      )}
      <Typo
        variant={TypoVariants.subTitle}
        color={TypoColors.greyMedium}
        className={styles.labelDisplayedRows}
      >
        {labelDisplayedRows(
          count === 0 ? 0 : (page - 1) * rowsPerPage + 1,
          Math.min(count, page * rowsPerPage),
          count,
        )}
      </Typo>
      <Button
        variant={ButtonVariants.Icon}
        disabled={1 === page}
        onClick={handleOnClickPreviousButton}
      >
        <ArrowLeft />
      </Button>
      <Button
        variant={ButtonVariants.Icon}
        disabled={page === maxPage}
        onClick={handleOnClickNextButton}
      >
        <ArrowRight />
      </Button>
    </div>
  );
};

export default TablePagination;
