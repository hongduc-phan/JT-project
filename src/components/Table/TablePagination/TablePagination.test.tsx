import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import {spy} from 'sinon';

import TablePagination from './index';
import {ArrowRight} from '../../Icons';

describe('<TablePagination />', () => {
  it('Should match table snapshot', () => {
    const pagi = (
      <TablePagination
        labelRowsPerPage="Items per page:"
        rowsPerPageOptions={[5, 10, 25]}
        page={1}
        count={200}
        rowsPerPage={10}
      />
    );
    const tree = renderer.create(pagi).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should trigger change value', () => {
    const cbChangePage = spy();
    const cbChangeRow = spy();

    const wrapper = mount(
      <TablePagination
        labelRowsPerPage="Items per page:"
        rowsPerPageOptions={[5, 10, 25]}
        page={1}
        count={200}
        rowsPerPage={10}
        onChangePage={cbChangePage}
        onChangeRowsPerPage={cbChangeRow}
      />,
    );

    wrapper.find(ArrowRight).simulate('click');

    expect(cbChangePage.calledWith(2)).toBeTruthy();

    wrapper.find('input').simulate('click');

    expect(
      wrapper
        .find('li')
        .at(1)
        .text(),
    ).toEqual('10');
    wrapper
      .find('li')
      .at(1)
      .simulate('click');

    expect(cbChangeRow.calledWith(10)).toBeTruthy();
  });
});
