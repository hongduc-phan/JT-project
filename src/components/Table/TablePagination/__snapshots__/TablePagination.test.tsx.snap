// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TablePagination /> Should match table snapshot 1`] = `
<div
  className="wrapper"
>
  <span
    className="typo subTitle greyMedium left"
  >
    Items per page:
  </span>
  <div
    className="root textFieldRoot"
  >
    <div
      className="textField normal active textFieldFocus textField"
      onClick={[Function]}
    >
      <span
        className="arrow textFieldArrow"
      >
        <svg
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M6.175 6.667L10 10.483l3.825-3.816L15 7.842l-5 5-5-5z"
              fill="currentColor"
              fillRule="nonzero"
            />
            <path
              d="M0 20h20V0H0z"
            />
          </g>
        </svg>
      </span>
      <div
        className="formGroup"
      >
        <div
          className="formLabel"
        >
          <input
            className="input"
            hidden={true}
            onBlur={[Function]}
            onFocus={[Function]}
            readOnly={true}
            value={10}
          />
          <div
            className="input inputSelect inputSelect"
          >
            
          </div>
        </div>
      </div>
    </div>
    
  </div>
  <span
    className="typo labelDisplayedRows subTitle greyMedium left"
  >
    1-10 of 200
  </span>
  <button
    className="typo wrap icon button inherit middle"
    disabled={true}
    onClick={[Function]}
  >
    <svg
      height="1em"
      viewBox="0 0 20 20"
      width="1em"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M13.333 13.825L9.517 10l3.816-3.825L12.158 5l-5 5 5 5z"
          fill="currentColor"
          fillRule="nonzero"
        />
        <path
          d="M0 20V0h20v20z"
        />
      </g>
    </svg>
  </button>
  <button
    className="typo wrap icon button inherit middle"
    disabled={false}
    onClick={[Function]}
  >
    <svg
      height="1em"
      viewBox="0 0 20 20"
      width="1em"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M6.667 6.175L10.483 10l-3.816 3.825L7.842 15l5-5-5-5z"
          fill="currentColor"
          fillRule="nonzero"
        />
        <path
          d="M20 0v20H0V0z"
        />
      </g>
    </svg>
  </button>
</div>
`;
